app/src/main/java/com/rkm/tasky/util/storage/implementation/SessionStorageImpl.kt

```diff
@@ -71,22 +43,21 @@ class SessionStorageImpl @Inject constructor(
 
     override suspend fun setSession(info: AuthInfo) {
         dataStore.edit { preferences ->
-            if (info.accessToken.isNotEmpty()) preferences[keyMap[SESSION_ACCESS_TOKEN_KEY]!!] =
-                encryptor.encrypt(info.accessToken)
-            if (info.refreshToken.isNotEmpty()) preferences[keyMap[SESSION_REFRESH_TOKEN_KEY]!!] =
-                encryptor.encrypt(info.refreshToken)
-            if (info.fullName.isNotEmpty()) preferences[keyMap[SESSION_USER_ID_KEY]!!] =
-                encryptor.encrypt(info.userId)
-            if (info.userId.isNotEmpty()) preferences[keyMap[SESSION_FULL_NAME_KEY]!!] =
-                encryptor.encrypt(info.fullName)
-            if (info.email.isNotEmpty()) preferences[keyMap[SESSION_EMAIL_KEY]!!] =
-                encryptor.encrypt(info.email)
+            preferences[keyMap[SESSION_AUTH_USER_KEY]!!] = encryptor.encrypt(authUserToString(info))
         }
     }
 
     override suspend fun clearSession() {
         dataStore.edit { preferences ->
-            preferences.clear()
+            preferences.remove(keyMap[SESSION_AUTH_USER_KEY]!!)
         }
     }
+
+    private fun authUserFromString(user: String): AuthInfo {
+        return Gson().fromJson(user, AuthInfo::class.java)
+    }
+
+    private fun authUserToString(user: AuthInfo): String {
+        return Gson().toJson(user)
+    }
```

Comment:
Something to be a bit careful about with saving JSON (I also did that) is when working with R8 for release build obfuscation. The JSON data relies on being saved with very specific field names, but if R8 obfuscated your app before, the field names will be very different. That won't be a problem until the user updates the app, but then it'll crash since you're trying to decode JSON obfuscated with a different pattern than the one from the new app.

The fix is to work with `SerializedName` fields for the class you're serializing here (`AuthInfo`) or to exclude these classes in your proguard rules.